packages/api/amplify_api_dart/test/amplify_api_config_test.dart

```diff
@@ -58,20 +61,28 @@ void main() {
 
 EndpointConfig createEndpointConfig(
   String endpoint, {
-  EndpointType type = EndpointType.graphQL,
+  ApiType type = ApiType.graphQL,
   APIAuthorizationType authorizationType = APIAuthorizationType.apiKey,
 }) {
   const region = 'us-east-1';
   const apiKey = 'abc-123';
-
-  final config = AWSApiConfig(
-    endpointType: type,
-    endpoint: endpoint,
-    region: region,
-    authorizationType: authorizationType,
-    apiKey: apiKey,
-  );
-
-  final endpointConfig = EndpointConfig('GraphQL', config);
+  late final ApiOutputs config;
+  if (type == ApiType.graphQL) {
+    config = DataOutputs(
+      url: endpoint,
+      awsRegion: region,
+      defaultAuthorizationType: authorizationType,
+      apiKey: apiKey,
+      authorizationTypes: [authorizationType],
+    );
+  } else {
+    config = RestApiOutputs(
+      url: endpoint,
+      awsRegion: region,
+      authorizationType: authorizationType,
+      apiKey: apiKey,
+    );
+  }
```

Comment:
*suggestion:* Use a switch statement. It will allow config to be final and will ensure that we are handling each case (if we added a new ApiType in the future, we would be alerted that this code needs to be updated with a switch statement)

```suggestion
  final config = switch(type) {
    ApiType.graphQL => DataOutputs(
      url: endpoint,
      awsRegion: region,
      defaultAuthorizationType: authorizationType,
      apiKey: apiKey,
      authorizationTypes: [authorizationType],
    ),
    ApiType.rest => RestApiOutputs(
      url: endpoint,
      awsRegion: region,
      authorizationType: authorizationType,
      apiKey: apiKey,
    )
  };
```

Reply:
+1. Can also add a default case to throw with more meaningful messaging.

Reply:
type is an enum and all the valuse are covered in the switch. hence adding a default case (_) cuase analyzer error: This case is covered by the previous cases.